tests/components/UserActions/ClickedDocumentList.spec.ts

```diff
@@ -117,6 +118,54 @@ describe('ClickedDocumentList', () => {
         });
     });
 
+    it('should not show the same query twice', () => {
+        const createComponentInside = sandbox.stub(Initialization, 'automaticallyCreateComponentsInsideResult');
+
+        const buildAction = (hash: string, i: number) => {
+            const document = Fake.createFakeResult();
+            document.uri = hash;
+            return new UserAction(UserActionType.Click, new Date(i), { origin_level_1: 'foo', uri_hash: document.uri }, document);
+        };
+
+        const TEST_DUPLICATE = [
+            buildAction('someQuery', 0),
+            buildAction('someQuery2', 1),
+            buildAction('someQuery2', 2),
+            buildAction('someQuery2', 3),
+            buildAction('someQuery', 4)
+        ];
+
+        const mock = Mock.advancedComponentSetup<ClickedDocumentList>(
+            ClickedDocumentList,
+            new Mock.AdvancedComponentSetupOptions(null, { userId: 'testuserId' }, env => {
+                fakeUserProfileModel(env.root, sandbox).getActions.returns(Promise.resolve(TEST_DUPLICATE));
+                return env;
+            })
+        );
+
+        return delay(() => {
+            // Expand the whole list of query.
+            const button = mock.cmp.element.querySelector<HTMLElement>('.coveo-more-less');
+            if (button) {
+                button.click();
+            }
+
+            const list = mock.env.element.querySelector<HTMLOListElement>('.coveo-list');
+            expect(list.childElementCount).toBe(new Set(TEST_DUPLICATE.map(action => action.raw.uri_hash)).size);
+
+            // Check that the order is respected.
+            const sortedActions = TEST_DUPLICATE.reverse().reduce(
+                (acc, action) => (!acc.find(existing => existing.raw.uri_hash === action.raw.uri_hash) ? [...acc, action] : acc),
```

Comment:
No, you're reproducing the logic of the code, thus the tests does not really test per se.
Hardcode the input and output test values and compare the actual output to the test output.

Reply:
hardcode ark, but why not 😐

Reply:
Well, let's think about what the value of a unit test. It should ensure that the 'black box' that is your code, when given a specific input give you back an expected output. If you repeat the black box in the tests, you're not unit-testing.

Reply:
It's not reproduce. It just happen that the reduction to sort thing out is the same.